# SYRIATEL CUSTOMER CHURN PREDICTION 

# BUSINESS UNDERSTANDING 

Since 2000, Syriatel has been the leading  Syrian mobile telecommunication company. The company believes in offering high quality products and services to it customers to meet the need of making life easier at a reasonable price. However, the company is facing a challenge of increased customer churn which has been observed for some period of time. The increased customer churn rates have negatively affected the company's revenue demanding a comprehesive understanding of the root causes. Hence for Syriatel to remain competitive within the Syrian telecommunication market, there is a need to identify the factors resulting in the increased customer churn rates and formulate proactive strategies to ensure they retain the active subscribers and attract new customers. 

This project provides a data-driven strategy to help the company caddress the customer churn challenge. The project aims at building classification models to predict the likelihood of a customer terminating business with the company. By implimenting this data-driven startegy, the company can easily anticipate the probably of customer stopping business with them and formulate strategies to increase the customer's satisfaction and end up retaining the customer earlier. This strategy can drive the company to a competitive edge within the market. The project  will help the company to proactively minimize the increased customer churn rates and provide better decision-making strategies. 

## Problem Statement

The Syrian mobile telecommunication market has become very competitive. As Syriatel navigates the competitive industry, it is facing a challenge of increased customer churn which has led to a decline in its sale revenue and brand loyalty. 

The goal of this project is to develop data-driven strategies to address the churn rate challenge. This project will develop classification models to forecast the chances of a customer leaving the comapny. The predictions made by the models will help the company adopt better strategies to help increase customer loyalty, satisfaction and to eventually increase customer retention rates. 

## Project Objectives 

1. Analyze Syriatel's churn  historical dataset
2. Identify customer churn patterns 
3. Develop two models to predict the probability of customer terminating the company's products and services. 
4. Provide data-driven recommendatioons and insights from the analysis and model prediction outcomes. 

## Conclusion 

By embracing this data- driven strategy, the created churn prediction models will be significant in minimizimg customer churn rates and increase consumer retention. By implimenting these machine learning technique and utilizing the given performance metrics, the company can proactively anticipate  customer churn risk and provide actionable insights. The strategies  boost the company towards sustainable growth, better market position with increased customer retention and satisfaction. 

# DATA UNDERSTANDING 

To uncover pattern's related to customer churn at Syriatel, will leverage a dataset available at Kaggle [Syriatel Dataset](https://www.kaggle.com/datasets/becksddf/churn-in-telecoms-dataset) . 

The dataset contains customer data such as :
1. Demographic and account Info 
     * Account length: How long the customer has been with the company in (month )
     * State and Area Code: The Geographic details
2.  Service Features and plans
     * International plan: whether the customer has international calling plans 
    * Voicemail plan: whether the customer has voicemail subscriptions
    * Number Vmail Messages: Remaining voicemail messages. 
3. Call & Usage Data : Features realted to customer calling habits 
    * Total Day/Eve/Night/Intl Minutes
    * Total Day/Eve/Night/Intl Calls
    * Total Day/Eve/Night/Intl Charge
    * Customer Service Calls
4. Targer/ Predictor : Churn 

Whether a customer is likely to churn based on their demographic,account infor, services plan and call usage metrics. 


### A Brief Overview of the Dataset 

In [9]:
import pandas as pd 
churn = pd.read_csv('Syriatel_dataset.csv')
churn.head()

state  account length  area code phone number international plan  \
0    KS             128        415     382-4657                 no   
1    OH             107        415     371-7191                 no   
2    NJ             137        415     358-1921                 no   
3    OH              84        408     375-9999                yes   
4    OK              75        415     330-6626                yes   

  voice mail plan  number vmail messages  total day minutes  total day calls  \
0             yes                     25              265.1              110   
1             yes                     26              161.6              123   
2              no                      0              243.4              114   
3              no                      0              299.4               71   
4              no                      0              166.7              113   

   total day charge  ...  total eve calls  total eve charge  \
0             45.07  ...               99             16.78   
1             27.47  ...              103             16.62   
2             41.38  ...              110             10.30   
3             50.90  ...               88              5.26   
4             28.34  ...              122             12.61   

   total night minutes  total night calls  total night charge  \
0                244.7                 91               11.01   
1                254.4                103               11.45   
2                162.6                104                7.32   
3                196.9                 89                8.86   
4                186.9                121                8.41   

   total intl minutes  total intl calls  total intl charge  \
0                10.0                 3               2.70   
1                13.7                 3               3.70   
2                12.2                 5               3.29   
3                 6.6                 7               1.78   
4                10.1                 3               2.73   

   customer service calls  churn  
0                       1  False  
1                       1  False  
2                       0  False  
3                       2  False  
4                       3  False  

[5 rows x 21 columns]

In [10]:
## Explore existing columns and row 
churn.shape

(3333, 21)

In [12]:
## View of the descriptive statistics 
churn.describe ()

account length    area code  number vmail messages  total day minutes  \
count     3333.000000  3333.000000            3333.000000        3333.000000   
mean       101.064806   437.182418               8.099010         179.775098   
std         39.822106    42.371290              13.688365          54.467389   
min          1.000000   408.000000               0.000000           0.000000   
25%         74.000000   408.000000               0.000000         143.700000   
50%        101.000000   415.000000               0.000000         179.400000   
75%        127.000000   510.000000              20.000000         216.400000   
max        243.000000   510.000000              51.000000         350.800000   

       total day calls  total day charge  total eve minutes  total eve calls  \
count      3333.000000       3333.000000        3333.000000      3333.000000   
mean        100.435644         30.562307         200.980348       100.114311   
std          20.069084          9.259435          50.713844        19.922625   
min           0.000000          0.000000           0.000000         0.000000   
25%          87.000000         24.430000         166.600000        87.000000   
50%         101.000000         30.500000         201.400000       100.000000   
75%         114.000000         36.790000         235.300000       114.000000   
max         165.000000         59.640000         363.700000       170.000000   

       total eve charge  total night minutes  total night calls  \
count       3333.000000          3333.000000        3333.000000   
mean          17.083540           200.872037         100.107711   
std            4.310668            50.573847          19.568609   
min            0.000000            23.200000          33.000000   
25%           14.160000           167.000000          87.000000   
50%           17.120000           201.200000         100.000000   
75%           20.000000           235.300000         113.000000   
max           30.910000           395.000000         175.000000   

       total night charge  total intl minutes  total intl calls  \
count         3333.000000         3333.000000       3333.000000   
mean             9.039325           10.237294          4.479448   
std              2.275873            2.791840          2.461214   
min              1.040000            0.000000          0.000000   
25%              7.520000            8.500000          3.000000   
50%              9.050000           10.300000          4.000000   
75%             10.590000           12.100000          6.000000   
max             17.770000           20.000000         20.000000   

       total intl charge  customer service calls  
count        3333.000000             3333.000000  
mean            2.764581                1.562856  
std             0.753773                1.315491  
min             0.000000                0.000000  
25%             2.300000                1.000000  
50%             2.780000                1.000000  
75%             3.270000                2.000000  
max             5.400000                9.000000

After loading the kaggle csv dataset to a Dataframe, it is clear that the dataset provides data on 3333 rows of customer and  21 columns of customer features. 
The dataset provides all the relevant features to model the customer churn prediction model covered later in this project . 

# DATA PREPARATION 